src/Component/Navbar.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useMemo, useState, useCallback } from "react";
+import "../App.css";
+import CurrencyData from "../data/currency-codes.json";
+import LanguageData from "../data/language-codes.json";
+import RegionData from "../data/regions.json";
+
+export const Navbar: React.FC = () => {
+  // State variables to track selected currency, language, region, and search query
+  const [selectedCurrency, setSelectedCurrency] = useState<string>("");
+  const [selectedLanguage, setSelectedLanguage] = useState<string>("");
+  const [selectedRegion, setSelectedRegion] = useState<string>("");
+  const [search, setSearch] = useState<string>("");
+
+  // Memoized options for currency selection
+  const CurrencyOptions = useMemo(() => {
```

Comment:
Generally, its not great practice to return a UI component from useMemo. It breaks react's rendering. React watches for the state that a component uses to update the UI. If it's inside a use memo, it will only update the UI if the state in the dependency array is updated. This makes it tricky to maintain. -3